### ISSUE TITLE
Upgrade Documentation - recommend more specific notes please!

### ISSUE THREAD

**Issue by kript** [irods/irods#2965]
I'm starting to look at this, and I'm not entirely au fait with V4, so apologies for what I am sure are going to be stupid questions, but I'm trying to come from a new upgrader's perspective and writing what I see missing in the documentation... :-)

In the upgrade manual;

There appears to be an assumption that the upgrade from run-in-place for 3.3.1 is going to a packaged installation. Is run-in-place to run-in-place supported, and if so, what are the changes to the procedure?

The document also appears to assume a move to JSON syntax. My memory (which may be faulty), was that JSON wasn't mandatory until 4.2?

Assuming that I did want to do a upgrade from run-in-place to packaged install, however;

*Step 2* says;
```
Make a backup of the iCAT database & configuration files: core.re, core.fnm, core.dvm, etc.
```
However, _etc_ covers a lot of sins! Can a full list of the files to be backed up be provided please?

*Step 10* says
```
If necessary, migrate 3.3.x in-place iCAT database to the system database installation.
```
This infers that iRODS 4.1.7 assumes the database is on the local host. This may not be what you mean. I'm assuming it means something like; "make a copy of your 3.3.1 iCAT database for your 4.1.7", and assumes that you will not be upgrading the database in-situ, which you may not be (and if this is contra-indicated, please provide reasons why (or link to github issues if a succinct summary isn't appropriate)

*Step 12* says
```
Manually update any changes to 'core.re' and 'server_config.json'. Keep in mind immediate replication rules (acPostProcForPut, etc.) may be superceded by your new resource composition
```
Update from what to what? At this point in the manual, the user may not be familiar enough with V4 to know what will break! Replication note is a good example, but a list of rules to be aware of would be very helpful. 

How does one convert the existing config to JSON, is there a guide or section in the manual? Searching for JSON didn't help - is it as simple as updating the server_config.json that is put into /etc/irods/server_config.json ? Which bits should be changed, and which can be kept the same? Any gotchas that could be referred to (github issues fine!)

*Step 13* says
```
Run ./packaging/setup_irods.sh (recommended) OR Manually update all 4.0+ configuration files given previous 3.3.x configuration (.irodsEnv, .odbc.ini DSN needs to be set to either 'postgres', 'mysql', or 'oracle'). The automatic ./packaging/setup_irods.sh script will work only with the system-installed database server.
```
Are .irodsEnv, .odbc.ini the only files that need to be changed? How would I know?
Again the reference to a system installed database server - does this mean locally?
Where are the reference notes for doing this with all the database plugins - have there been upgrade notes written for all of them (I'm particularly concerned with Oracle, of course, but if the default is postgres, then the MySQL users may also want this...

*Step 16* says
```
On all resource servers in the same Zone, install and setup 4.0+. Existing configuration details should be ported as well ('server.config', 'core.re', Vault permissions).
```
As above a list of files to update and pointers as to how to cover to JSON appreciated.

*Step 20* says
```
Sunset 3.3.x server(s)
```
I'm not sure what this means; didn't we upgrade the iCAT and all the iRES in steps 15 and 16?

Apologies for the stream of consciousness, please ask for clarification  as required!

**Comment by trel** [comment]
There are no plans to document a migration to 4.x with `--run-in-place` (https://github.com/irods/irods/issues/2968)

I'll work on a more specific list of configuration files for Step 2.

The database can be an external database in Step 10, will update that.

Steps 12 and 13 demand more attention, but specificity is hard there - every installation is different.

The configuration files are converted if the new JSON files are not found.  Once found, they will not be converted again.

Step 20 is just referring to the old code that was running in a regular user's namespace, usually with a local instance of a database.  This step is a placeholder for 'do whatever you want to delete the old installation, it is no longer needed'.  I see now that 'server' might be read as a machine, rather than a piece of software.  I'll update it.

**Comment by kript** [comment]
Thanks for all of this. I think I've got confused with the run-in-place and packaged installs because I don't see the packaged installs taking the run-in-place configs, that has to be done manually. I think this document refers to a run-in-place to run-in-place upgrade, or a packaged to packed upgrade. While I think this is entirely sensible it wasn't obvious to me starting down the road initially, so might be worth adding in a statement about what circumstances this upgrade doc is supposed to cover. Feel free to use the instructions I mailed Jason that we plan to use for upgrading from 3.3.1 to 4.x as a template for a once in a lifetime 3-4 migration doc, though!
